### PR TITLE
Gravity upgrade training initial commit

### PR DESCRIPTION
**Pull request by aelkugia**
Initial commit for Gravity upgrade training. I've added the following:

- Intro to workshop
- Upgrade overview (stepping through each of the steps at a high level)
- Placeholders for the additional sections we discussed (gravity upgrade status, logs and more + exploring a gravity manual upgrade + section for upgrade scenarios)

### CODE REVIEW

File: gravity_upgrade.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+2. Gravity uses k8s rolling update mechanism to perform the update.
+
+3. If applicable, before or after the upgrade custom update hooks can be used to perform application specific actions.
+
+Upgrades can be triggered both via the Gravity Hub (Ops Center) or from the CLI. Below we will go over the upgrade via the CLI to give some detail to the above steps mentioned.
+
+### Step 1: Providing new version of application
+
+As mentioned the first step in the upgrade process is to import your new application bundle onto a Gravity cluster and transfer it onto a cluster nodes to execute the upgrade command.
```
Comment (r0mant) [inline gravity_upgrade.md:25]: "application bundle" -> "cluster image" (bundle is an older term we no longer use)

Comment (r0mant) [inline gravity_upgrade.md:25]: Also, instead of "onto a cluster nodes" I would specify that it needs to be transferred on one of master nodes.

File: gravity_upgrade.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+Upgrades can be handled both in online and offline environments.
+
+#### Online Gravity Upgrade
+
+In the case a Gravity cluster is connected to the Gravity Hub (Ops Center) and it has remote support turned on, you can download an updated application bundle directly from the Ops Center.
```
Comment (r0mant) [inline gravity_upgrade.md:31]: "application bundle" -> "cluster image", "Ops Center" -> "Hub"

File: gravity_upgrade.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+#### Online Gravity Upgrade
+
+In the case a Gravity cluster is connected to the Gravity Hub (Ops Center) and it has remote support turned on, you can download an updated application bundle directly from the Ops Center. 
+
+Using the `gravity update download` you can check and download any updates
+
+#### Offline Gravity Upgrade
+
+In the case where a Gravity cluster is offline the new version of the application bundle will need to be copied over to one of the application cluster nodes that is accessible to the other nodes.
```
Comment (r0mant) [inline gravity_upgrade.md:37]: "application bundle" -> "cluster image", and same comment as above about the tarball placed on a master node.

File: gravity_upgrade.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+In order to upload to the new version you will need to extract the tarball and launch the `upload` script.
+
+### Step 2: Performing Gravity Upgrade
+
+As mentioned a Gravity upgrade can be triggered via the Gravity Hub (Ops Center) or from the CLI. In order to trigger the upgrade from the CLI you can extra the application bundle tarball to a directory and inside the execute the `upgrade` script.
```
Comment (r0mant) [inline gravity_upgrade.md:43]: extra -> extract, "cluster image"

File: gravity_upgrade.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+```
+tar xf application-bundle.tar
+ls -lh
+sudo ./gravity upgrade
```
Comment (r0mant) [inline gravity_upgrade.md:48]: After unpacking the tarball, you need to either execute the "upgrade" script (`sudo ./upgrade`), or "upload" script (`sudo ./upload`) followed by "gravity upgrade" command (`sudo ./gravity upgrade`). In your example you're only showing gravity upgrade command without upload. I think it might be worth mentioning both as sometimes there are cases you might want to only upload w/o triggering an upgrade.

File: gravity_upgrade.md [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+```
+#### Automatic Upgrade
+
+Following executing the upgrade command following the automatic upgrade process, the following steps will take place:
+
+1. Upgrade agent gets deployed on each cluster node as a `systemd` unit called `gravity-agent.service`
+
+`./gravity agent deploy`
```
Comment (r0mant) [inline gravity_upgrade.md:56]: Are these command here supposed to illustrate what the automatic upgrade is doing and show that those steps can be performed by users as well?

File: gravity_upgrade.md [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+
+#### Manual Upgrade
+
+Another option for a Gravity upgrade is a manual upgrade which we will cover by stepping through an example later in this workshop.
+
+In order to execute a manual upgrade the operation will be started by adding `--manual | -m flag` with the upgrade command:
+
+`sudo ./gravity upgrade --manual`
+
+To ensure version compatibility the manual upgrade process requires the Gravity binary to be included in the upgrade tarball.
```
Comment (r0mant) [inline gravity_upgrade.md:74]: I would rephrase this to say that to ensure version compatibility, all upgrade related commands (agent deployment, phase execution/rollback, etc.) need to be executed using the gravity binary included in the upgrade tarball.

File: gravity_upgrade.md [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+
+A manual upgrade operation starts with an operational plan which is a tree of actions required to be performed in the specified order to achieve the goal of the upgrade. This concept of the operation exists in order to have sets of smaller steps during an upgrade which can be re-executed or rolled back. Again, more on this as we will step through this in more detail further into the workshop.
```
Comment (r0mant) [inline gravity_upgrade.md:76]: operational -> operation

File: gravity_upgrade.md (hunk 6 above)
Comment (aelkugia) [reply gravity_upgrade.md:56]: Yeah the goal was to show what the automatic upgrade is doing, but can be confusing - I'll remove these
